### ISSUE TITLE
Fails to connect to localhost under Docker(Debian) on Windows

### ISSUE THREAD

**Issue by maxmah** [yhirose/cpp-httplib#894]
I'm running Debian-based Docker container ("mcr.microsoft.com/vscode/devcontainers/base:0-debian-10") under Windows and trying to connect to a localhost using cpp-httplib.

My HTTP server implementation is running under "localhost:3000".
In this scenario I'm supposed to use "http://host.docker.internal:3000" from inside the container when connecting to the host.
"curl" as well as "https://github.com/elnormous/HTTPRequest" library have no issues connecting and fetching a response, so I conclude that my server and docker-to-host connection work as expected.

When I'm trying to do a "Get" call to fetch the same data with the cpp-httplib I'm getting "Connection" error.
I traced it down to the call to "getaddrinfo(host, service.c_str(), &hints, &result);" inside the "create_socket" function.

The "getaddrinfo" gets correct input data of host/service as "http://host.docker.internal" and "3000", and it returns an error code "-2", which means after decoding: "Name or service not known", this in turn results in "INVALID_SOCKET" error in "create_socket" function.

It seems that "getaddrinfo" is not capable of processing "http://host.docker.internal:3000" in this case, while other libraries have no problems with this.

**Comment by yhirose** [comment]
@maxmah, thanks for the report. I have no clue why that happens.

https://github.com/yhirose/cpp-httplib/blob/6ff84d34d15c8cec51cd36246412b159bbb48484/httplib.h#L2069

Could you try to set `hints.ai_family` to something else like `AF_INET` or `AF_INET6` to see if it makes any difference?

**Comment by maxmah** [comment]
Thanks for the quick reply! Neither AF_INET nor AF_INET6 worked.

Curiously, after some more attempts I found a workaround. Somehow if I don't use "http://" it works, so:

```httplib::Client httpClient("host.docker.internal", 3000);  // works```

```httplib::Client httpClient("http://host.docker.internal", 3000);  // doesn't work```

**Comment by yhirose** [comment]
@maxmah, How about the following?

```cpp
httplib::Client httpClient("http://host.docker.internal:3000");
```

**Comment by maxmah** [comment]
Thanks a lot for the clarification! Your suggestion with `scheme_host_port` worked indeed, and I thought I tried it as well before creating a bug, but apparently I missed something (didn't recompile correctly?).

I looked more into the implementation, and indeed the 2 parameter constructor shouldn't have the "http://" prefix as it is not parsed and stripped before passing it further. Indeed the `getaddrinfo` will simply fail if it has the prefix and it expects a pure host name, so that explains everything (I guess I misinterpreted the documentation). 

Thanks again for the quick support!
